src/App.tsx

```diff
@@ -1,56 +1,71 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { SuggestionPeople } from './components/suggestionPeople';
+import { Person } from './types/Person';
+import { getSuggestedPeople } from './helpers/getSuggestedPeople';
+import { QuerySuggestions } from './components/querySuggestions';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+const debounce = (
+  func: (value: string) => void,
+  delay: number,
+) => {
+  let timerId: number;
+
+  return (...args: string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
+
+interface AppProps {
+  delay?: number,
+}
+
+export const App: React.FC<AppProps> = ({ delay = 1000 }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
```

Comment:
let's pass our delay directly to the function
```suggestion
    debounce(setAppliedQuery, 1000),
```